executor/executor_linux.h

```diff
@@ -248,3 +251,26 @@ static feature_t features[] = {
     {"usb", setup_usb},
     {"sysctl", setup_sysctl},
 };
+
+static void setup_machine()
+{
+	// nmi_check_duration() prints "INFO: NMI handler took too long" on slow debug kernels.
+	// It happens a lot in qemu, and the messages are frequently corrupted
+	// (intermixed with other kernel output as they are printed from NMI)
+	// and are not matched against the suppression in pkg/report.
+	// This write prevents these messages from being printed.
+	// Note: this is not executed in C reproducers.
+	if (!write_file("/sys/kernel/debug/x86/nmi_longest_ns", "10000000000"))
+		printf("write to /sys/kernel/debug/x86/nmi_longest_ns failed: %s\n", strerror(errno));
+}
+
+static void dFetch_enable()
+{
+	int fd = open("/sys/kernel/debug/dfetch_detection", O_RDWR);
+	if (fd == -1)
+		fail("open of /sys/kernel/debug/dfetch_detection failed");
+	if (ioctl(fd, DFETCH_ENABLE))
```

Comment:
Does this enable double fetch detection per-process or per-thread?
Please add a link to corresponding kernel support to the commit description.